stdlib/n-ary-app.kl

```diff
@@ -109,23 +109,47 @@
                                           (cons-list-syntax 'my-arrow args stx))
                                         stx))])]))]))
 
-(export (rename ([my-arrow ->]) my-arrow))
+(meta
+  (define optional-parens
+    (lambda (symbol)
+      (lambda (stx)
+        (syntax-case stx
+          [(ident x)
+           (pure (list-syntax (symbol) stx))]
+          [(cons x xs)
+           (pure (cons-list-syntax symbol xs stx))])))))
+
+-- TODO: wrap 'datatype' so that parentheses are also optional for
+-- the nullary types and constructors it creates
+(define-macros
+  ([my-Unit (optional-parens 'Unit)]
+   [my-unit (optional-parens 'unit)]
+   [my-Bool (optional-parens 'Bool)]
+   [my-true (optional-parens 'true)]
+   [my-false (optional-parens 'false)]
+   [my-nothing (optional-parens 'nothing)]))
+
+(export (rename ([my-arrow ->]
+                 [my-Unit Unit] [my-unit unit]
+                 [my-Bool Bool] [my-true true] [my-false false]
+                 [my-nothing nothing])
+                my-arrow
+                my-Unit my-unit
+                my-Bool my-true my-false
+                my-nothing))
 (export #%module #%app
         lambda define example define-macros quote meta
         the Syntax Macro
         with-unknown-type
-        if true false
```

Comment:
Apparently we were previously exporting `true` and `false` twice, should that be an error?

At some point I was both exporting `my-true` as true in the above `export (rename ...)` and `true` as `true` here, thus leading to mysterious errors. So I think exporting different values with the same name should definitely be an error.

Reply:
Yeah, sounds like an error to me. #74